lib/netext/httpext/request.go

```diff
@@ -314,6 +315,11 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 	mreq := preq.Req.WithContext(ctx)
 	res, resErr := client.Do(mreq)
 
+	if res != nil && res.StatusCode == http.StatusSwitchingProtocols {
```

Comment:
Can you add a `TODO` for fixing this properly (i.e. not relying on the status, but on whether the body is write-able), linking to the underlying go issue that's preventing us from doing so right now: https://github.com/golang/go/issues/31391